internal/app/machined/pkg/runtime/v1alpha1/platform/gcp/metadata.go

```diff
@@ -61,16 +79,26 @@ func (g *GCP) getMetadata(context.Context) (*MetadataConfig, error) {
 		return nil, err
 	}
 
-	if meta.PublicIPv4, err = metadata.ExternalIP(); err != nil {
-		if _, ok := err.(metadata.NotDefinedError); !ok {
-			return nil, err
-		}
-	}
-
 	meta.Preempted, err = metadata.Get("instance/scheduling/preemptible")
 	if err != nil {
 		return nil, err
 	}
 
 	return &meta, nil
 }
+
+func (g *GCP) getNetworkMetadata(ctx context.Context) ([]NetworkInterfaceConfig, error) {
+	metadataNetworkConfigDl, err := download.Download(ctx, GCPInterfacesEndpoint,
```

Comment:
can we use the Google metadata package for it? 

Looks like we could with `metadata.Get("instance/network-interfaces/?recursive=true&alt=json")` and decode the result as JSON. Let's not use this ugly custom download requests :)